utils/s2n_random.c

```diff
@@ -542,38 +553,46 @@ static int s2n_rand_init_cb_impl(void)
     return S2N_SUCCESS;
 }
 
+bool s2n_supports_custom_rand()
+{
+#if !defined(S2N_LIBCRYPTO_SUPPORTS_ENGINE)
+    return false;
+#endif
+
+    return !s2n_libcrypto_is_boringssl() && !s2n_libcrypto_is_libressl() && !s2n_libcrypto_is_awslc() && !s2n_libcrypto_is_fips();
```

Comment:
What's the difference between `s2n_is_in_fips_mode` and `s2n_libcrypto_is_fips`? We were gating functionality on `s2n_libcrypto_is_fips` on line 552 and we don't have that anymore. Just wanting to make sure the logic is the same after this change.

Reply:
I see. So it's essentially caching the value. Let's use `s2n_is_in_fips_mode` then.

Reply:
Changed in https://github.com/aws/s2n-tls/pull/4878/commits/75d49ab93d490ac31f4a9d87fde4770afbc3db69

I also hid `s2n_libcrypto_is_fips` since that really should be used once at initialization time.